File: Blind75/Arrays&Hashing/src/GroupAnagram.java

import java.util.*;

class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap<String , List<String>> hmap = new HashMap<>();
        
        for (String s : strs)
        {
            int[] count = new int[26];
            for ( char c : s.toCharArray())
            {
                count[c-'a']++;
            }
            String key = Arrays.toString(count);
            hmap.putIfAbsent(key, new ArrayList<String>());
            hmap.get(key).add(s);
        }
        return new ArrayList<>(hmap.values());
        
    }
}

public class GroupAnagram{
    public static void main(String[] args) {
        Solution sol = new Solution();
        String[] strs = {"eat","tea","tan","ate","nat","bat"};
        
        System.out.println(sol.groupAnagrams(strs));

    }
}

File: Blind75/Python/Arrays&Hashing/GroupAnagram.py
from typing import List


class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        res ={}
        for word in strs:
            count = []
            for c in word:
                count[ord(c)-ord('a')]=+1
            res[tuple(count)].append(word)
        return list[res.values()]